src/main/resources/org/jenkinsci/modules/windows_slave_installer/jenkins-slave.xml

```diff
@@ -46,4 +50,22 @@ THE SOFTWARE.
   <logmode>rotate</logmode>
 
   <onfailure action="restart" />
+  
+  <!-- 
+    In the case WinSW gets terminated and leaks the process, we want to abort
+    these runaway JAR processes on startup to prevent "Slave is already connected errors" (JENKINS-28492).
+  -->
+  <extensions>
+    <!-- This is a sample configuration for the RunawayProcessKiller extension. -->
+    <extension enabled="true" 
+               className="winsw.Plugins.RunawayProcessKiller.RunawayProcessKillerExtension"
+               id="killOnStartup">
+      <pidfile>%BASE%\jenkins_agent.pid</pidfile>
+      <stopTimeout>5000</stopTimeout>
```

Comment:
I will keep 5 seconds for now.
We have a windows service startup timeout (~30 sec), and the logic must complete before it.
Here we are killing runaway processes, hence force-kill after the timeout should be fine.

Before the final release I will test it anywat